controllers/awsmachine_controller.go

```diff
@@ -157,6 +158,8 @@ func (r *AWSMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reter
 
 	// Always close the scope when exiting this function so we can persist any AWSMachine changes.
 	defer func() {
+		// set Ready condition before AWSMachine is patched
+		conditions.SetSummary(machineScope.AWSMachine)
```

Comment:
Wondering if it make sense to express the Summary with the "1 of x task completed" notation, e.g. https://github.com/kubernetes-sigs/cluster-api/blob/c437644da00406b515201a48b5b93f29b09d3878/test/infrastructure/docker/controllers/dockermachine_controller.go#L120

Reply:
SGTM will update